PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,31 @@
+(Describe your pull request in 1-2 sentences)
+
+Fix # (optional)
+
+Rationale and detailed description of the changes:
+
+-
+-
+-
+
+Tasks before review:
+
+- [ ] Add tests and examples of any added functionality
+    - [ ] For bugfix pull requests, list any previously-disabled tests that are
+          now passing (and any tests to catch similar bugs elsewhere in the
+          code)
+- [ ] Run `make lint` on the project, ensure it passes
+    - [ ] Run `make pretty-cpp` and `make pretty-python`, if necessary
+    - [ ] Review variable names, make sure they're descriptive (ask a friend)
+    - [ ] Review all TODOs, convert to issues wherever possible
+    - [ ] Make sure draft, in-progress, and commented-out code is moved to its
+          own branch
+- [ ] If committing any code in Jupyter/IPython notebooks, install and run `nbstripout`
+- [ ] Merge with master and resolve any conflicts
+- [ ] (anything else that's still in progress)
```

Comment:
I would remove this item, in favor of the list below

Reply:
It's a template for contributors to add other tasks that need to be finished _before_ the PR can be considered ready for review and merging. This contrasts with the list below, which is for tasks that are out-of-scope and left to another PR.

Reply:
ok, that was somehow unclear to me. Could we then add a requirement that any task left out for future PR gets an issue? This will ensure they are remembered.